include/boost/math/tools/polynomial.hpp

```diff
@@ -134,7 +236,11 @@ class polynomial
 
    // access:
    size_type size()const { return m_data.size(); }
-   size_type degree()const { return m_data.size() - 1; }
+   size_type degree()const
+   { 
+       BOOST_ASSERT(size() != 0);
```

Comment:
On thinking about it further, I would actually recommend against throwing an exception. If a user calls `degree()` on a zero polynomial, it's tantamount to calling `front()` on an empty `vector`: it's simply a logic bug. The user should be checking whether a polynomial is zero and, knowing the mathematics of polynomials, not asking what the degree of the zero polynomial is, just as we don't divide by zero and expect forgiveness. If it were to throw an exception, `logic_error` would be most appropriate, the user now has to check for that exception, which is probably less productive than if they properly checked for the zero polynomial value in the first place.

Now, to counter my own argument: `size()` actually provides exactly the right interface because it's defined for all polynomials including our problematic friend the zero polynomial. We don't really need `degree()`, but we have it, so we can make it a bit inefficient with the reassurance that we can always use `size()` for the most consistent and efficient interface.

So, I don't mind particularly, but I am inclined towards the first option because I think the second option fosters a lack of rigour in thinking about arithmetic. In the end, it's your call. :)